ext/objspace/object_tracing.c

```diff
@@ -70,18 +73,41 @@ delete_unique_str(st_table *tbl, const char *str)
     }
 }
 
+static void
+light_newobj_i(VALUE tpval, void *data)
+{
+    struct traceobj_arg *arg = (struct traceobj_arg *)data;
+    rb_trace_arg_t *tparg = rb_tracearg_from_tracepoint(tpval);
+    unsigned int locindex = rb_tracearg_locindex(tparg);
+    st_data_t obj = (st_data_t)rb_tracearg_object(tparg);
+
+    RUBY_ASSERT(st_lookup(arg->object_table, obj, NULL) == 0);
+
+    if (locindex) {
+        st_add_direct(arg->object_table, obj, (st_data_t)locindex);
+    }
+}
+
+static void
+light_freeobj_i(VALUE tpval, void *data)
+{
+    struct traceobj_arg *arg = (struct traceobj_arg *)data;
+    rb_trace_arg_t *tparg = rb_tracearg_from_tracepoint(tpval);
+    st_data_t obj = (st_data_t)rb_tracearg_object(tparg);
+
+    st_delete(arg->object_table, &obj, NULL);
+}
+
 static void
 newobj_i(VALUE tpval, void *data)
 {
     struct traceobj_arg *arg = (struct traceobj_arg *)data;
     rb_trace_arg_t *tparg = rb_tracearg_from_tracepoint(tpval);
     VALUE obj = rb_tracearg_object(tparg);
-    VALUE path = rb_tracearg_path(tparg);
-    VALUE line = rb_tracearg_lineno(tparg);
+    unsigned int locindex = rb_tracearg_locindex(tparg);
```

Comment:
`locindex` name is unclear IMHO. Maybe `location_index` or `line_index`?

Reply:
it is intentional to make a mintage to separate general concept.

Reply:
To the casual observer it's not obvious what `locindex` is.